tests/flows/teleporter_registry.go

```diff
@@ -0,0 +1,221 @@
+package flows
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	runner_sdk "github.com/ava-labs/avalanche-network-runner/client"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp/payload"
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	teleporterregistry "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/upgrades/TeleporterRegistry"
+	"github.com/ava-labs/teleporter/tests/interfaces"
+	"github.com/ava-labs/teleporter/tests/utils"
+	deploymentUtils "github.com/ava-labs/teleporter/utils/deployment-utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	networkID              uint32 = 1337
+	teleporterByteCodeFile        = "./contracts/out/TeleporterMessenger.sol/TeleporterMessenger.json"
+)
+
+func TeleporterRegistry(network interfaces.LocalNetwork) {
```

Comment:
IIUC, this test only updates the Teleporter version for subnets A and B. Won't that leave the subnets incompatible with subnet C and the C-Chain for any tests that run after this? 

In any case, can you add a comment to the function description explaining the state changes made by this test? It seems possible that a future test may assume a certain Teleporter address, which would change after this test.

Reply:
yup that's a good point. Right now after this test runs the C-Chain's `TeleporterRegistry` will be behind on a Teleporter version. The reason tests are still okay for now is because the tests use C-Chain as source mostly, so the destination dapp would still receive from an older version. 

Currently the `subnet-evm` release is not compatible with a `coreth` that has Warp off-chain, will need to update this after so that all subnets match in registered Teleporter versions.

Reply:
I like that we use the same registry and have random order of e2e tests, so we can that the after updating the registry across all chains, the other e2e tests all still work. Will just need to add the new protocol version to C-chain after a new compatible release is made.